src/extensions/bigbang/bigbang.go

```diff
@@ -523,3 +494,155 @@ func findImagesforBBChartRepo(repo string, values chartutil.Values) (images []st
+			"./zarf tools kubectl wait gitrepositories.source.toolkit.fluxcd.io bigbang -n bigbang --for jsonpath={.status.conditions[0].observedGeneration}=$GENERATION\n",
+		Mute:       &t,
+		MaxRetries: &maxRetries,
+	})
+
+	c.Actions.OnDeploy.After = append(c.Actions.OnDeploy.After, types.ZarfComponentAction{
+		Description:     fmt.Sprintf("Big Bang GitRepository to be ready"),
+		MaxTotalSeconds: &maxTotalSeconds,
+		Wait: &types.ZarfComponentActionWait{
+			Cluster: &types.ZarfComponentActionWaitCluster{
+				Kind:       "gitrepository.source.toolkit.fluxcd.io",
+				Identifier: "bigbang",
+				Namespace:  ns,
+				Condition:  "ready",
+			},
+		},
+	})
+
+	c.Actions.OnDeploy.After = append(c.Actions.OnDeploy.After, types.ZarfComponentAction{
+		Description: "Resuming bigbang-bigbang HelmChart",
+		Cmd: "./zarf tools kubectl patch helmcharts.source.toolkit.fluxcd.io bigbang-bigbang -n bigbang --type json -p '[{\"op\": \"add\", \"path\": \"/spec/suspend\", \"value\": false}]'\n" +
+			"GENERATION=$(./zarf tools kubectl get helmcharts.source.toolkit.fluxcd.io bigbang-bigbang -n bigbang -o jsonpath={.metadata.generation})\n" +
+			"./zarf tools kubectl wait helmcharts.source.toolkit.fluxcd.io bigbang-bigbang -n bigbang --for jsonpath={.status.conditions[0].observedGeneration}=$GENERATION\n",
+		Mute:       &t,
+		MaxRetries: &maxRetries,
+	})
+
+	c.Actions.OnDeploy.After = append(c.Actions.OnDeploy.After, types.ZarfComponentAction{
+		Description:     fmt.Sprintf("Big Bang HelmChart to be ready"),
+		MaxTotalSeconds: &maxTotalSeconds,
+		MaxRetries:      &maxRetries,
+		Wait: &types.ZarfComponentActionWait{
+			Cluster: &types.ZarfComponentActionWaitCluster{
+				Kind:       "helmcharts.source.toolkit.fluxcd.io",
+				Identifier: "bigbang-bigbang",
+				Namespace:  ns,
+				Condition:  "ready",
+			},
+		},
+	})
+
+	c.Actions.OnDeploy.After = append(c.Actions.OnDeploy.After, types.ZarfComponentAction{
+		Description: "Resuming bigbang HelmRelease",
+		Cmd: "./zarf tools kubectl patch helmrelease bigbang -n bigbang --type json -p '[{\"op\": \"add\", \"path\": \"/spec/suspend\", \"value\": false}]'\n" +
+			"GENERATION=$(./zarf tools kubectl get helmrelease bigbang -n bigbang -o jsonpath={.metadata.generation})\n" +
+			"./zarf tools kubectl wait helmrelease bigbang -n bigbang --for jsonpath={.status.observedGeneration}=$GENERATION\n",
+		Mute:       &t,
+		MaxRetries: &maxRetries,
+	})
+
+	c.Actions.OnDeploy.After = append(c.Actions.OnDeploy.After, types.ZarfComponentAction{
+		Description:     fmt.Sprintf("Big Bang HelmRelease to be ready"),
+		MaxTotalSeconds: &maxTotalSeconds,
+		MaxRetries:      &maxRetries,
+		Wait: &types.ZarfComponentActionWait{
+			Cluster: &types.ZarfComponentActionWaitCluster{
+				Kind:       "HelmRelease",
+				Identifier: "bigbang",
+				Namespace:  ns,
+				Condition:  "ready",
+			},
+		},
+	})
+
+	for _, hrNamespacedName := range namespacedHelmReleaseNames {
```

Comment:
The resume being a part of the loop (vs parallel resume) is going to artificially inject a lot of lag time in my opinion. While a lot of BB things have dependencies there are also a lot that should be able to reconcile in parallel. This structure is effectively causing a 1 at a time reconciliation as far as I can tell?

Reply:
Yeah I think you're right. I wrote that trying to be really generic instead of waiting for, say, the `monitoring` HR to finish. Could def shore up the logic and make it more efficient